source/Octopus.Server.Client/Operations/RegisterKubernetesClusterOperation.cs

```diff
@@ -14,20 +13,29 @@ public RegisterKubernetesClusterOperation(IOctopusClientFactory clientFactory) :
     {
     }
 
+    public string DefaultNamespace { get; set; }
+
+
     protected override void PrepareMachineForReRegistration(MachineResource machine, string proxyId)
     {
         machine.Endpoint = GenerateEndpoint(proxyId);
     }
 
     protected override EndpointResource GenerateEndpoint(string proxyId)
     {
-        return CommunicationStyle switch
+        var endpoint = CommunicationStyle switch
         {
             CommunicationStyle.TentaclePassive => new KubernetesTentacleEndpointResource(
                 new ListeningTentacleEndpointConfigurationResource(TentacleThumbprint, GetListeningUri()) { ProxyId = proxyId }),
             CommunicationStyle.TentacleActive => new KubernetesTentacleEndpointResource(
                 new PollingTentacleEndpointConfigurationResource(TentacleThumbprint, SubscriptionId.ToString())),
             _ => null
         };
+
+        if (endpoint is null) return null;
```

Comment:
Do we ever expect this operation to be called with other communication styles? Are we better off throwing at this point?

Reply:
Good point, I've made it throw an exception 👍